hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkFileFormatInternalRowReaderContext.scala

```diff
@@ -87,17 +88,15 @@ class SparkFileFormatInternalRowReaderContext(parquetFileReader: SparkParquetRea
     }
     val structType = HoodieInternalRowUtils.getCachedSchema(requiredSchema)
     if (FSUtils.isLogFile(filePath)) {
-      val projection = HoodieInternalRowUtils.getCachedUnsafeProjection(structType, structType)
-      new CloseableMappingIterator[InternalRow, UnsafeRow](
+      val dataSchemaWithMergeCol = if (hasRowIndexField) {
+        HoodiePositionBasedSchemaHandler.addPositionalMergeCol(dataSchema)
+      } else {
+        dataSchema
```

Comment:
Should this use `requiredSchema`?

Reply:
The HoodieSparkFileReaderFactory parquet reader doesn't have schema evolution implemented so we want to read using the write schema because we know we can do that without any errors. But now that you point it out, we might be reading a lot of columns we don't need. So need to think more about what to do here

Reply:
I see.

Reply:
In most cases we still need to read all columns for merging so this may be OK.  We should file a follow-up Jira to see how to fix the projection properly.

Reply:
The rewrite I did now has evolution inside the HoodieSparkFileReaderFactory parquet reader. So now we pass in the required schema